client/src/components/Login.vue

```diff
@@ -38,40 +41,46 @@ axios.defaults.withCredentials = true;
 export default {
   data() {
     return {
+      valid: false,
       response: [],
       errors: [],
-
       user: {
         username: "",
         password: ""
-      }
+      },
+      userNameRules: [v => !!v || "Name is required"],
+      passwordRules: [v => !!v || "Password is required"]
     };
   },
 
   methods: {
     login() {
-      axios
-        .post(
-          `http://localhost:3000/api/users/login`,
-          {
-            username: this.user.username,
-            password: this.user.password
-          },
-          { withCredentials: true }
-        )
-        .then(response => {
-          // JSON responses are automatically parsed.
-          if (response.status == "200") {
-            // success
-            window.localStorage.setItem("token", response.data);
-            window.location = "/";
-          } else {
-            // invalid credentials
-          }
-        })
-        .catch(e => {
-          this.errors.push(e);
-        });
+      axios.defaults.withCredentials = true;
+      if (this.$refs.form.validate()) {
+        this.valid = true;
+
+        axios
+          .post(
+            `http://localhost:3000/api/users/login`,
+            {
+              username: this.user.username,
+              password: this.user.password
+            },
+            { withCredentials: true }
+          )
+          .then(response => {
+            if (response.status == "200") {
+              window.location = "/dashboard";
+            } else {
+              // invalid credentials
```

Comment:
moet hier iets?

Reply:
Daar zou een error message moeten komen, maar wanneer je een incorrecte username/password invult gaat de request heel lang op pending staan. Hierdoor is het nog niet mogelijk om op een korte tijd een error aan de gebruiker te laten zien. Ik moet later een manier zoeken die de repsonse time inkort zodat ik iets met die response data kan doen.